shared/domain/src/main/java/com/ivy/domain/usecase/BalanceBuilder.kt

```diff
@@ -0,0 +1,63 @@
+package com.ivy.domain.usecase
+
+import com.ivy.data.model.primitive.AssetCode
+import com.ivy.data.model.primitive.NonZeroDouble
+import com.ivy.data.model.primitive.PositiveDouble
+
+class BalanceBuilder {
+
+    private val balance = mutableMapOf<AssetCode, NonZeroDouble>()
+
+    fun processDeposit(
```

Comment:
This namings are good 💯 nice!

Reply:
Let's rename it to `processDeposits` and `processWithdrawals` (plural because we're processing more than once thing)